rst_files/kalman.rst

```diff
@@ -666,29 +619,22 @@ Exercise 1
 
 .. code-block:: julia
 
-    using Distributions
-
     # parameters
-
     θ = 10
     A, G, Q, R = 1.0, 1.0, 0.0, 1.0
     x_hat_0, Σ_0 = 8.0, 1.0
 
     # initialize Kalman filter
-
     kalman = Kalman(A, G, Q, R)
     set_state!(kalman, x_hat_0, Σ_0)
-    # == Run == #
 
-    N = 5
     xgrid = range(θ - 5, θ + 2, length = 200)
-    densities = []
-    labels = []
-    for i ∈ 1:N
+    densities = Array{Array{Float64, 1}, 1}(undef, 5) # we'll do 5 rounds of updating
+    labels = ["t=1", "t=2", "t=3", "t=4", "t=5"]
+    for i ∈ 1:5 
         # Record the current predicted mean and variance, and plot their densities
         m, v = kalman.cur_x_hat, kalman.cur_sigma
-        push!(densities, pdf.(Normal(m, sqrt(v)), xgrid))
-        push!(labels, LaTeXString("\$t=$i\$"))
+        densities[i] = pdf.(Normal(m, sqrt(v)), xgrid)
```

Comment:
Instead of a pre-allocation and loop, why not a list comprehension.

Reply:
Because this is a sequential computation (as opposed to strictly iterating over some static object, which can be done asynchronously), I think the logic as clearer as a loop.